src/pages/settings/Profile/Contacts/NewContactMethodPage.js

```diff
@@ -52,99 +52,84 @@ const defaultProps = {
     loginList: {},
 };
 
-class NewContactMethodPage extends Component {
-    constructor(props) {
-        super(props);
-
-        this.state = {
-            login: '',
-            password: '',
-        };
-        this.onLoginChange = this.onLoginChange.bind(this);
-        this.validateForm = this.validateForm.bind(this);
-        this.submitForm = this.submitForm.bind(this);
-    }
-
-    onLoginChange(login) {
-        this.setState({login});
-    }
+function NewContactMethodPage(props) {
+    const [login, setLogin] = useState('');
+    const [password, setPassword] = useState('');
+    const loginInputRef = useRef(null);
 
     /**
     * Determine whether the form is valid
     *
     * @returns {Boolean}
     */
-    validateForm() {
-        const login = this.state.login.trim();
-        const phoneLogin = LoginUtils.getPhoneNumberWithoutSpecialChars(login);
+    const validateForm = useCallback(() => {
```

Comment:
NAB, since we are calling this `validateForm()` method in the render... I'd maybe update this to be a `useMemo()` and return the value instead of a function... 

```js
const isFormValid = useMemo(() => {...}, dependencies);
```

It's kind of minor, but I think slightly better in that we don't need to call the function on render and can just use the value directly instead.

Reply:
yeah i was really confused looking at this and was thinking it'd be better if it was a single value not a function but then didn't want to accidentally break something. can update though.